lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -186,10 +198,13 @@ def refund(money, intent_id, options = {})
       def store(payment_method, options = {})
         params = {}
         post = {}
-
         # If customer option is provided, create a payment method and attach to customer id
         # Otherwise, create a customer, then attach
-        if payment_method.is_a?(StripePaymentToken) || payment_method.is_a?(ActiveMerchant::Billing::CreditCard)
+        if payment_method.is_a?(ActiveMerchant::Billing::NetworkTokenizationCreditCard)
```

Comment:
🛑  Warning:

Seems that we have a big issue here, tokenization and storing payment methods are different things on Stripe, tokenization is just a temporary storage (One time usage) for a particular payment method and vaulting a payment method on stripe terms means save payment method into a customer object to be use later.

So here we are adding code to just tokenized a payment method to be used only one time (just for the particular payment intent), we should move this code into the authorize/purchase flow where it belongs.